pkg/bmc/access_test.go

```diff
@@ -11,162 +11,192 @@ func init() {
 }
 
 func TestParseLibvirtURL(t *testing.T) {
-	T, H, P, A, err := getTypeHostPort("libvirt://192.168.122.1:6233/")
+	url, err := getParsedURL("libvirt://192.168.122.1:6233/?abc=def")
```

Comment:
Rather than change every one of these tests it might be easier to just define a local `getTypeHostPort()` function.

Reply:
I want to convert all of these tests to be table-driven anyway, but I can do that in a separate PR. If I do that first, you'll have to rebase this, but you'll have fewer renames.

Reply:
I can take care of modifying them to be table-driven at the same time if you wish.

Reply:
See #300

Reply:
I have now modified the tests to be table-driven @dhellmann.

Reply:
Sorry, I missed this, I'll rebase on it and take it further

Reply:
@maelk I went ahead and merged the other change. That should make reviewing your rebase a little easier. And thanks for the additional changes you made in that file. I like the way you cleaned up the other tests, too.

Reply:
@dhellmann Thanks for the merge, I have now rebased this PR that should now be ready for review. Thank you